alf/networks/critic_networks.py

```diff
@@ -94,6 +94,12 @@ def __init__(self,
             raise ValueError(
                 'Only a single action is supported by this network')
 
+        if action_spec.is_discrete:
```

Comment:
discrete can still be used, right? For high dimensional discrete action, QNetwork may not be the best choice.

Reply:
If action spec is discrete then it has to go through a preprocessor first. Which is not supported for now. Right now the EncodingNetwork `self._action_encoder` will complain the shape is not correct.